ground/gcs/src/plugins/config/configmodulewidget.cpp

```diff
@@ -620,21 +618,46 @@ void ConfigModuleWidget::updateAirspeedGroupbox(UAVObject *obj)
     ui->gb_airspeedGPS->setChecked(false);
     ui->gb_airspeedPitot->setChecked(false);
 
-    switch (airspeedSettingsData.AirspeedSensorType) {
-    case AirspeedSettings::AIRSPEEDSENSORTYPE_GPSONLY:
+    if (airspeedSettingsData.AirspeedSensorType == AirspeedSettings::AIRSPEEDSENSORTYPE_GPSONLY) {
         ui->gb_airspeedGPS->setChecked(true);
-        break;
-    case AirspeedSettings::AIRSPEEDSENSORTYPE_DIYDRONESMPXV5004:
-        ui->cb_pitotType->setCurrentIndex(AirspeedSettings::AIRSPEEDSENSORTYPE_DIYDRONESMPXV5004);
-        ui->gb_airspeedPitot->setChecked(true);
-    case AirspeedSettings::AIRSPEEDSENSORTYPE_DIYDRONESMPXV7002:
-        ui->cb_pitotType->setCurrentIndex(AirspeedSettings::AIRSPEEDSENSORTYPE_DIYDRONESMPXV7002);
-        ui->gb_airspeedPitot->setChecked(true);
-    case AirspeedSettings::AIRSPEEDSENSORTYPE_EAGLETREEAIRSPEEDV3:
-        ui->cb_pitotType->setCurrentIndex(AirspeedSettings::AIRSPEEDSENSORTYPE_EAGLETREEAIRSPEEDV3);
-        ui->gb_airspeedPitot->setChecked(true);
-        break;
     }
+    else {
+         ui->gb_airspeedPitot->setChecked(true);
+    }
+}
+
+/**
+ * @brief ConfigModuleWidget::toggleAirspeedType Toggle the airspeed sensor type based on checkbox
+ * @param checked
+ */
+void ConfigModuleWidget::enableAirspeedTypeGPS(bool checked)
+{
+    if (checked){
+        AirspeedSettings *airspeedSettings;
+        airspeedSettings = AirspeedSettings::GetInstance(getObjectManager());
+        AirspeedSettings::DataFields airspeedSettingsData;
+        airspeedSettingsData = airspeedSettings->getData();
+        airspeedSettingsData.AirspeedSensorType = AirspeedSettings::AIRSPEEDSENSORTYPE_GPSONLY;
+        airspeedSettings->setData(airspeedSettingsData);
```

Comment:
Fortunately this isn't the case; the `clicked` signal only gets emitted when the user clicks the checkbox, not when it is changed using `setChecked`.